homeassistant/components/asuswrt/__init__.py

```diff
@@ -77,10 +81,29 @@
         dnsmasq=conf[CONF_DNSMASQ],
     )
 
-    await api.connection.async_connect()
+    try:
+        await api.connection.async_connect()
+    except OSError as ex:
```

Comment:
We need to be able to distinguish bad credentials from bad connection. Only bad connection should spawn the retry mechanism. Bad credentials should log error and `return False` since it's a non recoverable error.

Reply:
All errors that are due to user configuring bad config are also non recoverable.

Reply:
A config flow would help here since we can check the connection during set up and let the user know before finishing the config.

Reply:
In any case if the user change credential on router the component will fail during startup. In config flow should only asked a limited set of parameters and remaining should be managed in configuration.yaml (for expert use).
I will start to review the code to retry only in case of OSError exception and return false if is_connected report false.

Reply:
Credentials will normally be part of the config flow, unless they are client secrets in an oauth flow.

If the credentials change we can handle that, during set up and launch a new config flow and the user will be notified to complete it again. The only requirement is that we can distinguish bad credentials from bad connection.

Reply:
Config flow is out of scope for this PR. :+1: